client/state/sharing/keyring/selectors.js

```diff
@@ -49,6 +56,50 @@ export function getUserConnections( state, userId ) {
 }
 
 /**
+ * Given a service, returns a flattened array of all possible accounts for the
+ * service for which a connection can be created.
+ *
+ * @param  {Object} state   Global state tree
+ * @param  {String} service The name of the service to check
+ * @return {Array}          Flattened array of all possible accounts for the service
+ */
+export function getAvailableExternalConnections( state, service ) {
+	const siteUserConnectionsForService = getSiteUserConnectionsForService(
+		state, getSelectedSiteId( state ), getCurrentUserId( state ), service
+	);
+
+	// Iterate over Keyring connections for this service and generate a
+	// flattened array of all accounts, including external users
+	return flatten( getKeyringConnectionsByName( state, service ).map( ( keyringConnection ) => {
```

Comment:
Noting that this was copied from existing code, but I feel a `Array#reduce` would be a much better fit for what we're trying to achieve here rather than `flatten( Array#map )`, i.e.:

```js
return getKeyringConnectionsByName( state, service ).reduce( ( memo, keyringConnection ) => {
	return memo.concat( [ {
		name: keyringConnection.external_display || keyringConnection.external_name,
		picture: keyringConnection.external_profile_picture,
		keyringConnectionId: keyringConnection.ID,
		isConnected: some( siteUserConnectionsForService, {
			keyring_connection_ID: keyringConnection.ID,
			external_ID: keyringConnection.external_ID,
		} )
	} ].concat( keyringConnection.additional_external_users.map( ( externalUser ) => ( {
		ID: externalUser.external_ID,
		name: externalUser.external_name,
		picture: externalUser.external_profile_picture,
		keyringConnectionId: keyringConnection.ID,
		isConnected: some( siteUserConnectionsForService, {
			keyring_connection_ID: keyringConnection.ID,
			external_ID: externalUser.external_ID,
		} ),
		isExternal: true
	} ) ) ) );
}, [] );
```

The `concat`ing in this example is maybe a little less readable. Also, there's quite a bit of repeated logic in the object creation that could perhaps be split out.